src/inheritAttributes.js

```diff
@@ -50,17 +50,48 @@ export const buildBaseUrls = (referenceUrls, baseUrlElements) => {
  */
 export const getSegmentInformation = (adaptationSet) => {
   const segmentTemplate = findChildren(adaptationSet, 'SegmentTemplate')[0];
-  const segmentTimeline =
-    segmentTemplate && findChildren(segmentTemplate, 'SegmentTimeline')[0];
   const segmentList = findChildren(adaptationSet, 'SegmentList')[0];
+  const segmentUrls = segmentList && findChildren(segmentList, 'SegmentURL')
+    .map(s => shallowMerge({ tag: 'SegmentURL' }, getAttributes(s)));
   const segmentBase = findChildren(adaptationSet, 'SegmentBase')[0];
+  const segmentTimelineNode = segmentList || segmentTemplate;
+  const segmentTimeline =
+    segmentTimelineNode && findChildren(segmentTimelineNode, 'SegmentTimeline')[0];
+  const segmentInitializationNode = segmentList || segmentBase || segmentTemplate;
```

Comment:
These variable names are a bit misleading. `segmentTimelineNode` and `segmentInitializationNode` are actually the parent nodes of those nodes